controller/search_blackbox_test.go

```diff
@@ -63,6 +66,135 @@ func (s *searchControllerTestSuite) SetupTest() {
 	s.controller = NewSearchController(s.svc, s.GormDB, spaceBlackBoxTestConfiguration)
 }
 
+func (s *searchControllerTestSuite) TestSearchWorkItemsCSV() {
+	deserialize := func(csvStr string) ([]map[string]string, error) {
+		if csvStr == "" || csvStr == "\n" {
+			return []map[string]string{}, nil
+		}
+		var result []map[string]string
+		reader := csv.NewReader(strings.NewReader(csvStr))
+		// parse header line
+		keys, err := reader.Read()
+		if err == io.EOF {
```

Comment:
I think we can get rid of this `EOF` check and do `if err != nil`  We should return an error even if it's not `io.EOF` error.

Reply:
But EOF is also an error, so I'd need that check anyway to see if it is the end of the file or a "real" error..?

Reply:
Yes. If both the error have to be dealt differently, we should have two error checks.

Reply:
You lost me :) The EOF is not an error per se. If we get to an EOF, we need to break out of the loop and return the result.

Reply:
So we have 
```go
keys, err := reader.Read()
if err == io.EOF {
 ...
}

for {
...
}
```

What happens if the error is not `EOF` ? what if the read failed due to some other reason? We aren't dealing with those errors. 

Does it  make sense now?

Reply:
Ahh, sorry, I was focussing on the loop area. Yes, you're right.

Reply:
Fixed 6b5e31a.